### ISSUE TITLE
Drugs Orphaned from Ingredient

### ISSUE THREAD

**Issue by ericaVoss** [OHDSI/Vocabulary-v5.0#136]
Related to issue #63.  Tested on Vocab 20170503.

We have found Drugs that are not associated to their ingredient.  This will cause problems in DRUG_ERA building or locating items of a specific ingredient.

```{SQL}
WITH CTE_CONCEPT_ANCESTOR AS (
	SELECT c.CONCEPT_ID, c.CONCEPT_NAME, c.CONCEPT_CLASS_ID, c.INVALID_REASON,
		c2.CONCEPT_ID AS DESCENDANT_CONCEPT_ID, c2.CONCEPT_NAME AS DESCENDANT_CONCEPT_NAME, 
		c2.CONCEPT_CLASS_ID AS DESCENDANT_CONCEPT_CLASS_ID, c2.INVALID_REASON AS DESCENDANT_INVALID_REASON
	FROM CONCEPT c
		JOIN CONCEPT_ANCESTOR ca
			ON ca.DESCENDANT_CONCEPT_ID = c.CONCEPT_ID
		JOIN CONCEPT c2
			ON c2.CONCEPT_ID = ca.ANCESTOR_CONCEPT_ID
	WHERE c.DOMAIN_ID = 'DRUG'
	AND c.STANDARD_CONCEPT = 'S'
), 
CTE_INGREDIENT_ONLY AS (
	SELECT *
	FROM CTE_CONCEPT_ANCESTOR
	WHERE DESCENDANT_CONCEPT_CLASS_ID = 'Ingredient'
)
SELECT DISTINCT ca1.CONCEPT_ID, ca1.CONCEPT_NAME, ca1.CONCEPT_CLASS_ID, ca1.INVALID_REASON, 
	ca2.DESCENDANT_CONCEPT_ID, ca2.DESCENDANT_CONCEPT_NAME, ca2.DESCENDANT_CONCEPT_CLASS_ID, ca2.DESCENDANT_INVALID_REASON
FROM CTE_CONCEPT_ANCESTOR ca1
	LEFT OUTER JOIN CTE_INGREDIENT_ONLY ca2
		ON ca1.CONCEPT_ID = ca2.CONCEPT_ID
WHERE ca2.DESCENDANT_CONCEPT_ID IS NULL
ORDER BY ca1.CONCEPT_NAME
```

One specific examples we care about, this concept gets used in the data but doesn't get associated to the ingredient.
```
45777080	{3 (0.8 ML adalimumab 50 MG/ML Prefilled Syringe [Humira]) } Pack [Humira Pediatric Crohn's Disease Starter Package (3 count)]
```

Credit to @fdefalco  & @AnthonyMolinaro for finding the issue.

Tagging @clairblacketer & @alondhe

**Comment by aostropolets** [comment]
I checked your query on the latest vocabulary version, didn't return anything.
Also 45777080 has adalimumab in concept_ancestor.
Seems like you simply need to download the August release.

**Comment by ericaVoss** [comment]
@aostropolets thanks!  I will try that!

**Comment by AnthonyMolinaro** [comment]
@aostropolets Hi Anna, thanks for the update.  Since we're not on the new vocab yet, can you tell us more about this issue?  We'd like to understand how this might occur (we suspect the process that creates concept_ancestor from concept_relationship might be the cause).  Also, was this something that was specifically fixed in the August release or do you just not happen to see the problem in the August release?  Thanks.

**Comment by aostropolets** [comment]
@AnthonyMolinaro 
We were missing automatically generated links between more granualar and more general concepts in the RxNorm due to a bug in the script. And hence the Ancestor Constructor wasn't able to create those relationships between them.

**Comment by ericaVoss** [comment]
@aostropolets - Looks much better on Vocabulary 20170920!

I still get 433 rows that return with the above query.  I think a lot of them maybe shouldn't go anywhere (like tricky vaccines), but might be worth a review.  Here are some examples:

```{TEXT}
43011779	{7 (24 HR Rotigotine 0.0833 MG/HR Transdermal System) / 7 (24 HR Rotigotine 0.167 MG/HR Transdermal System) } Pack
40213271	Bacillus Calmette-Guerin vaccine
2718655		Laetrile, amygdalin, vitamin b17
42620870	laureth-9 Injection
```

**Comment by aostropolets** [comment]
@ericaVoss 
First of all, non-standard concepts don't make it into the concept_ancestor. We've recently made RxNorm Extension out of DPD so it became non-standard. You also have updated/deprecated concepts like 43587477	Lisinopril 5.44 MG [Lisinopril-Ratiopharm]	  that we don't proceed.
Missing drugs: most of them shouldn't actually exist like all those Amino Acids. 
CPT4 and HCPCS don't have ingredients in their hierarchies.
The bottom line is, we have CVX orphans. Need to investigate them more.

**Comment by ericaVoss** [comment]
@aostropolets question - what are CVX orphans?

**Comment by aostropolets** [comment]
@ericaVoss 
CVX drugs that do not have ingredient in their hierarchy :) Seems to be a vocabulary processing bug.

**Comment by AnthonyMolinaro** [comment]
@aostropolets 
Hi Anna, I'm new to the vocab, can you explain what a cvx drug is?  Specifically, how to identify it with a query (when I check domain_name, concept_class_name, and concept_class_id I don't find the string "cvx").  Thanks!

**Comment by AnthonyMolinaro** [comment]
@aostropolets 
Ok, so I found CVX here: https://github.com/OHDSI/Vocabulary-v5.0/blob/master/Final_Assembly/PCONCEPTANCESTOR.prc

Line 261 has a case statement that includes,  dc.vocabulary_id='CVX',  so it seems CVX is a vocab identifier.   However, both of these queries:
select count(*) from vocabulary_20170503.dbo.vocabulary where lower(vocabulary_id) like '%cvx%'
select count(*) from vocabulary_20170503.dbo.concept where lower(vocabulary_id) like '%cvx%'

return 0.  Can you tell us a bit more about CVX.  Also, is that oracle stored proc the way that CONCEPT_ANCESTOR gets built?

Thanks!

**Comment by aostropolets** [comment]
@AnthonyMolinaro 
CVX is a standard drug vocabulary that contains vaccines.
You don't see it in 20170503 vocab as we just added it in the end of July.
Concept_ancestor: not sure what's the question. ancestor is built based on concept_relationship so you get all those relationships between concepts as hierarchical tree. If you are interested here the link:
https://github.com/OHDSI/Vocabulary-v5.0/blob/master/Final_Assembly/PCONCEPTANCESTOR.prc
May be quite hard to get the logic though.

**Comment by ericaVoss** [comment]
Testing 20180609

@aostropolets, I think this problem is solved now?
